sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -2117,6 +2117,198 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSQLContext {
     assert(ex4.getMessage.contains("data type mismatch: argument 3 requires int type"))
   }
 
+  test("transform keys function - test various primitive data types combinations") {
```

Comment:
We don't need so many cases here. We only need to verify the api works end to end.
Evaluation checks of the function should be in `HigherOrderFunctionsSuite`.